src/features/impl_alloc.rs

```diff
@@ -30,7 +29,21 @@ impl VecWriter {
 
 impl enc::write::Writer for VecWriter {
     fn write(&mut self, bytes: &[u8]) -> Result<(), EncodeError> {
-        self.inner.extend_from_slice(bytes);
+        self.inner.try_reserve(bytes.len())?;
+
+        let start = self.inner.len();
+
+        // Get a slice of `&mut [MaybeUninit<u8>]` of the remaining capacity
+        let remaining = &mut self.inner.spare_capacity_mut()[..bytes.len()];
```

Comment:
Is there a reason the second half of this function isn't just `self.inner.extend_from_slice(bytes)`? Since we already reserved it should be fine right? Or is `extend_from_slice` allowed to over-reserve?

Reply:
That makes sense :+1: updated